persistent/Database/Persist/Sql.hs

```diff
@@ -39,6 +42,24 @@ import Database.Persist.Sql.Orphan.PersistQuery
 import Database.Persist.Sql.Orphan.PersistStore
 import Database.Persist.Sql.Orphan.PersistUnique ()
 
+-- | Begin an SQL transaction.
+--
+-- @since 2.10.2
+transactionBegin :: MonadIO m => ReaderT SqlBackend m ()
```

Comment:
This needs to be generalized to `(MonadIO m, BackendCompatible SqlBackend backend) => ReaderT backend m ()`.